src/librustc_resolve/check_unused.rs

```diff
@@ -305,6 +307,31 @@ pub fn check_crate(resolver: &mut Resolver<'_>, krate: &ast::Crate) {
 
         let len = spans.len();
         spans.sort();
+        for sp in &spans {
+            debug!(". {:?}", sp);
+            // Do not emit unused warnings about failed use statements (#48244)
+            for &crate::PrivacyError(dedup_span, _, _) in &visitor.privacy_errors {
+                debug!("- {:?}", dedup_span);
+                if sp.overlaps(dedup_span) {
+                    debug!("break");
+                    break 'unused_imports;
```

Comment:
Wouldn't this get rid of *all* unused import errors in the case that *any* import failed? (that is to say, I'd have expected this to be `continue 'unused_imports`)

Reply:
Indeed. I made a few other changes, including replacing `break` for `continue`. (This was leftover from wip.)